npm/metrics/ai-utils.go

```diff
@@ -0,0 +1,93 @@
+package metrics
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-container-networking/aitelemetry"
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/Azure/azure-container-networking/npm/util"
+)
+
+var (
+	th aitelemetry.TelemetryHandle
+)
+
+// CreateTelemetryHandle creates a handler to initialize AI telemetry
+func CreateTelemetryHandle(version, aiMetadata string) error {
+
+	aiConfig := aitelemetry.AIConfig{
+		AppName:                   util.AzureNpmFlag,
+		AppVersion:                version,
+		BatchSize:                 util.BatchSizeInBytes,
+		BatchInterval:             util.BatchIntervalInSecs,
+		RefreshTimeout:            util.RefreshTimeoutInSecs,
+		DebugMode:                 util.DebugMode,
+		GetEnvRetryCount:          util.GetEnvRetryCount,
+		GetEnvRetryWaitTimeInSecs: util.GetEnvRetryWaitTimeInSecs,
+	}
+
+	var err error
+	for i := 0; i < util.AiInitializeRetryCount; i++ {
+		th, err = aitelemetry.NewAITelemetry("", aiMetadata, aiConfig)
+		if err != nil {
+			log.Logf("Failed to init AppInsights with err: %+v for %d time", err, i+1)
+			time.Sleep(time.Minute * time.Duration(util.AiInitializeRetryInMin))
+		} else {
+			i = util.AiInitializeRetryCount
```

Comment:
in Go it's more idiomatic to `break` from the for loop

Reply:
Nice! Break makes sense and it's a general practice for other languages too.